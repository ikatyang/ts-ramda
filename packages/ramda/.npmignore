node_modules/
scripts/
src/
templates/
.npmignore
test.ts
tsconfig.json
tslint.json
