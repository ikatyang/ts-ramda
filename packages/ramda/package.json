{
  "name": "@ikatyang/types-ramda",
  "version": "0.23.0",
  "description": "TypeScript's type definitions for Ramda",
  "repository": "https://github.com/ikatyang/types",
  "author": "ikatyang",
  "license": "MIT",
  "scripts": {
    "ts-node": "../../node_modules/.bin/ts-node -P ./tsconfig.json",
    "build": "yarn run ts-node -- ./scripts/build.ts ./templates ./typings"
  },
  "devDependencies": {
    "ramda": "0.23.0"
  }
}
