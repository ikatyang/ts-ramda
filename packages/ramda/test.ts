/// <reference path="./index.d.ts" />

import * as R from 'ramda';

R.__;

R.add(1, 2);
R.add(1)(2);
