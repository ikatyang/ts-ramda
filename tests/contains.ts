import * as R from 'ramda';

// @dts-jest
R.contains(3)([1, 2, 3]); // => true
// @dts-jest
R.contains(3, [1, 2, 3]); // => true
// @dts-jest
R.contains(4)([1, 2, 3]); // => false
// @dts-jest
R.contains({})([{}, {}]); // => false
const obj = {};
// @dts-jest
R.contains(obj)([{}, obj, {}]); // => true
