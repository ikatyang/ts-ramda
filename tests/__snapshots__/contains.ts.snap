// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`R_contains(string) 1`] = `"<T$1 extends string = string>(list: T$1[] | ArrayLike<T$1>) => boolean"`;

exports[`R_contains(string, string) 1`] = `"boolean"`;

exports[`R_contains(string, string_array) 1`] = `"boolean"`;
